### ISSUE TITLE
Import on bucket uses provider project instead of actual bucket project

### ISSUE THREAD

**Issue by toadjaune** [hashicorp/terraform-provider-google#8977]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```
Terraform v0.14.10
+ provider registry.terraform.io/hashicorp/google v3.65.0
```

### Affected Resource(s)

* `google_storage_bucket`

Tests were made using the resource, the corresponding datasource is likely affected too, but untested.

### Terraform Configuration Files

```tf
provider "google" {
  project     = "project1"
  credentials = "/some/path/to/key.json"
}

resource "google_storage_bucket" "example" {
  name    = "my-bucket-name"
  project = "project2"
}
```

### Panic Output

N/A

### Expected Behavior

The bucket should be imported in the state with the project field set to whatever project it exists in.

### Actual Behavior

The bucket is imported in the state with project "project1"

### Steps to Reproduce

* Manually create a GCS bucket with all default settings
* Put the above config in a .tf file (adapt auth method and bucket name, but leave both "project" definitions as is.)
* `terraform import import google_storage_bucket.example my-bucket-name`
* Open `terraform.tfstate` with any text editor, and see that .resources[0].instances[0].attributes.project is set to "project1", instead of the project id where your bucket is located.

Interestingly enough, only the "project" defined in the provider causes an issue. Doing the same test without a provider-level value results in the resource-level provider definition being (correctly) ignored, and the correct value being set.

### Important Factoids

N/A

### References

None that I could find

**Comment by venkykuberan** [comment]
@toadjaune tf import command ignores the project attribute for the resource being imported. It just looks for empty config.

Where does the bucket exist ? project1 or project2 ?

`Doing the same test without a provider-level value results in the resource-level provider definition being (correctly) ignored, and the correct value being set.`

Can you please clarify what's correct value being set ?

**Comment by toadjaune** [comment]
The bucket exists in neither, it's in a third project (let's call it `project3`)

With the config example given above, it's being set to `project1` (the one defined in the provider), instead of `project3` when importing.

On the other hand, funnily, it's being set to the correct value (`project3`) with the following config : 
```
provider "google" {
  credentials = "/some/path/to/key.json"
}

resource "google_storage_bucket" "example" {
  name    = "my-bucket-name"
  project = "project2"
}
```

**Comment by esciara** [comment]
Also `terraform apply -refresh-state` does not solve the issue: it simply keeps the project that has been set during import. Makes you wonder where the provider gets the information about the bucket, as it is not refreshed when refreshing the state file.

This also means that if a bucket has been manually moved from one project to another, `terraform` will not pick up the move and will still force deletion and recreation of the bucket.

A dirty work around is to do a search/replace on the buckets in the state file... Can't see another way to do it if you have buckets in different projets that are defined under the same provider (which has to be associated to a single project).

**Comment by dubschro** [comment]
Just ran into this today myself.  Still feels like a real issue despite the documentation saying it will infer from the provider block if present, which is what is happening in my case.  Feels like it should never infer and should always use the compute api and get authoritative information, especially for a globally scoped identifier such as buckets.

**Comment by gurusai-voleti** [comment]
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_bucket#import

Storage buckets can be imported using the name or project/name. If the project is not passed to the import command it will be inferred from the provider block or environment variables. If it cannot be inferred it will be queried from the Compute API (this will fail if the API is not enabled).

as per the documentation it is working fine now

**Comment by toadjaune** [comment]
I agree with @dubschro ; even if at least now the import behavior is documented, it's quite counter-intuitive, I'd expect the in-tfstate state of the resource after an import to reflect the actual object state, regardless of my configuration at import-time.

So, the issue as originally reported is still present in my opinion.

**Comment by gurusai-voleti** [comment]
> I agree with [@dubschro](https://github.com/dubschro) ; even if at least now the import behavior is documented, it's quite counter-intuitive, I'd expect the in-tfstate state of the resource after an import to reflect the actual object state, regardless of my configuration at import-time.
> 
> So, the issue as originally reported is still present in my opinion.

this is particularly for bucket not for objects, can you provide an non working example

**Comment by gurusai-voleti** [comment]
Please verify with latest version and let us know if reproducible

**Comment by toadjaune** [comment]
Hi,

I don't currently have any GCP-hosted environment anymore (switched companies since opening this issue) :/

By any chance, could someone else interested in this issue (ping @esciara @dubschro) quickly try to reproduce the issue on their end, please ? 🙏
